applications/CompressiblePotentialFlowApplication/custom_elements/transonic_perturbation_potential_flow_element.cpp

```diff
@@ -0,0 +1,856 @@
+    const int wake = r_this.GetValue(WAKE);
+
+    if (wake == 0) // Normal element (non-wake) - eventually an embedded
+        CalculateRightHandSideNormalElement(rRightHandSideVector, rCurrentProcessInfo);
+    else // Wake element
+        CalculateRightHandSideWakeElement(rRightHandSideVector, rCurrentProcessInfo);
+}
+
+template <int Dim, int NumNodes>
+void TransonicPerturbationPotentialFlowElement<Dim, NumNodes>::CalculateLeftHandSide(
+    MatrixType& rLeftHandSideMatrix, ProcessInfo& rCurrentProcessInfo)
+{
+    const TransonicPerturbationPotentialFlowElement& r_this = *this;
+    const int wake = r_this.GetValue(WAKE);
+
+    if (wake == 0) // Normal element (non-wake) - eventually an embedded
+        CalculateLeftHandSideNormalElement(rLeftHandSideMatrix, rCurrentProcessInfo);
+    else // Wake element
+        CalculateLeftHandSideWakeElement(rLeftHandSideMatrix, rCurrentProcessInfo);
+}
+
+template <int Dim, int NumNodes>
+void TransonicPerturbationPotentialFlowElement<Dim, NumNodes>::EquationIdVector(
+    EquationIdVectorType& rResult, ProcessInfo& CurrentProcessInfo)
+{
+    const TransonicPerturbationPotentialFlowElement& r_this = *this;
+    const int wake = r_this.GetValue(WAKE);
+
+    if (wake == 0) // Normal element
+    {
+        if (rResult.size() != NumNodes)
+            rResult.resize(NumNodes, false);
+
+        const int kutta = r_this.GetValue(KUTTA);
+
+        if (kutta == 0)
+            GetEquationIdVectorNormalElement(rResult);
+        else
+            GetEquationIdVectorKuttaElement(rResult);
+    }
+    else // Wake element
+    {
+        if (rResult.size() != 2 * NumNodes)
+            rResult.resize(2 * NumNodes, false);
+
+        GetEquationIdVectorWakeElement(rResult);
+    }
+}
+
+template <int Dim, int NumNodes>
+void TransonicPerturbationPotentialFlowElement<Dim, NumNodes>::GetDofList(DofsVectorType& rElementalDofList,
+                                                                 ProcessInfo& CurrentProcessInfo)
+{
+    const TransonicPerturbationPotentialFlowElement& r_this = *this;
+    const int wake = r_this.GetValue(WAKE);
+
+    if (wake == 0) // Normal element
+    {
+        if (rElementalDofList.size() != NumNodes)
+            rElementalDofList.resize(NumNodes);
+
+        const int kutta = r_this.GetValue(KUTTA);
+
+        if (kutta == 0)
+            GetDofListNormalElement(rElementalDofList);
+        else
+            GetDofListKuttaElement(rElementalDofList);
+    }
+    else // wake element
+    {
+        if (rElementalDofList.size() != 2 * NumNodes)
+            rElementalDofList.resize(2 * NumNodes);
+
+        GetDofListWakeElement(rElementalDofList);
+    }
+}
+
+template <int Dim, int NumNodes>
+void TransonicPerturbationPotentialFlowElement<Dim, NumNodes>::FinalizeSolutionStep(ProcessInfo& rCurrentProcessInfo)
+{
+    bool active = true;
+    if ((this)->IsDefined(ACTIVE))
+        active = (this)->Is(ACTIVE);
```

Comment:
Some comments,
```suggestion
        const bool active = Is(ACTIVE);
```
- The `IsDefined` check is not required in this case (you can't arrive to this point without defining the `ACTIVE` flag as it is required by the Builder and Solver).
- All the elements are `ACTIVE` by default. If this is not changed outside the element, this check shouldn't be required.